make_release/bump-version.nu

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env nu
+use std log
+
+# bump the minor or patch version of a Rust / Nushell project
+def main [
+    --patch: bool  # update the minor version instead of the minor
+]: nothing -> nothing {
+    let version = open Cargo.toml
+        | get package.version
+        | parse "{major}.{minor}.{patch}"
+        | into int major minor patch
+        | into record
+
+    let new_version = if $patch {
+        $version | update patch { $in + 1 }
+    } else {
+        $version | update minor { $in + 1 } | update patch { 0 }
+    }
+
+    let version = $version | transpose | get column1 | str join "."
+    let new_version = $new_version | transpose | get column1 | str join "."
+
+    log info $"bumping all packages and Nushell files in (open Cargo.toml | get package.name) from ($version) to ($new_version)"
+
+    ls **/Cargo.toml | each {|file|
+        log debug $"bumping ($file.name) from ($version) to ($new_version)"
+        open --raw $file.name | str replace --all --string $'version = "($version)"' $'version = "($new_version)"' | save --force $file.name
+    }
+    ls **/*.nu | each {|file|
```

Comment:
I was wondering if this was a good opportunity to somehow use
```nushell
❯ "default_{config,env}.nu" | str expand
╭───┬───────────────────╮
│ 0 │ default_config.nu │
│ 1 │ default_env.nu    │
╰───┴───────────────────╯
```
since I think those are the only 2 nushell files that need updating, instead of listing the 50+ nu files in the repo.

Reply:
i think this is a good idea and a good occasion to use `str expand` :blush:

Reply:
It just seems to me that trying to do a string replace in every single .nu file is wasteful and could be problematic at some point. I'd prefer to limit it to the files you want to change. It's kind of the same idea as earlier in the script where you use Cargo.toml versus *.toml. You're ensuring that you're only looking at the files you want to change.

Reply:
should be good to go with 7af23d0 :yum: